### ISSUE TITLE
`MgetResponseItem` documentation inaccurate

### ISSUE THREAD

**Issue by gannett-ggreer** [elastic/go-elasticsearch#749]
https://pkg.go.dev/github.com/elastic/go-elasticsearch/v8@v8.11.0/typedapi/types#MgetResponseItem

Documentation says:
```
MgetResponseItem holds the union for the following types:

GetResult
MultiGetError
```
but the type is declared as `interface{}` so when the `json.Decode` is called the underlying type becomes `map[string]interface{}`, which is not very helpful in the "typed" API and makes using a `mget.Response` rather difficult.

**Comment by Anaethelion** [comment]
Agreed that `map[string]interface{}` is not ideal and lives as a default for now.

The `MgetResponseItem` union lacks a type to switch upon to identify its kind. The current plan for the client is to have some lookahead within the deserialization to identify exclusive fields and use that to determine which type to deserialize into.

**Comment by Anaethelion** [comment]
Fixed in the aforementioned PR, this will be part of the next release!
